samples/notebooks/quickstart.ipynb

```diff
@@ -0,0 +1,566 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2.1 Create a Docker container\n",
+    "Create your own container image that includes your program. If your component creates some outputs to be fed as inputs to the downstream components, each separate output must be written as a string to a separate local text file by the container image. For example, if a trainer component needs to output the trained model path, it can write the path to a local file `/output.txt`. The string written to an output file cannot be too big. If it is too big (>> 100 kB), save the output to an external persistent storage and pass the storage path to the next component.\n",
+    "\n",
+    "Start by entering the value of your Google Cloud Platform Project ID."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following cell creates a file `app.py` that contains a Python script. The script takes a GCS bucket name as an input argument, gets the lists of blobs in that bucket, prints the list of blobs and also writes them to an output file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# Create folders if they don't exist.\n",
+    "mkdir -p tmp/components/list-gcs-blobs\n",
+    "\n",
+    "# Create the Python file that lists GCS blobs.\n",
+    "cat > ./tmp/components/list-gcs-blobs/app.py <<HERE\n",
+    "import argparse\n",
+    "from google.cloud import storage\n",
+    "# Parse agruments.\n",
+    "parser = argparse.ArgumentParser()\n",
+    "parser.add_argument(\n",
+    "    '--bucket', type=str, required=True, help='GCS bucket name.')\n",
+    "args = parser.parse_args()\n",
+    "# Create a client.\n",
+    "storage_client = storage.Client()\n",
+    "# List blobs.\n",
+    "bucket = storage_client.get_bucket(args.bucket)\n",
+    "list_blobs_response = bucket.list_blobs()\n",
+    "blobs = ','.join([blob.name for blob in list_blobs_response])\n",
+    "print(blobs)\n",
+    "with open('/blobs.txt', 'w') as f:\n",
+    "  f.write(blobs)\n",
+    "HERE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now create a container that runs the script. Start by creating a `Dockerfile`. A `Dockerfile` contains the instructions to assemble a Docker image. The `FROM` statement specifies the Base Image from which you are building. `WORKDIR` sets the working directory. When you assemble the Docker image, `COPY` will copy the required files and directories (for example, `app.py`) to the filesystem of the container. `RUN` will execute a command (for example, install the dependencies) and commits the results. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# Create Dockerfile.\n",
```

Comment:
The recommended way to build a component image is through the Lambda SDK(https://github.com/kubeflow/pipelines/blob/d2f5cc92a46012b9927209e2aaccab70961582dc/sdk/python/kfp/compiler/_component_builder.py#L488)

For example: https://github.com/kubeflow/pipelines/blob/master/samples/notebooks/KubeFlow%20Pipeline%20Using%20TFX%20OSS%20Components.ipynb

Reply:
Wanted to show users how to create an image.